pkg/csvcopy/csvcopy.go

```diff
@@ -142,6 +145,32 @@ func (c *Copier) Copy(ctx context.Context, reader io.Reader) (Result, error) {
 		}
 	}
 
+	// Setup reader with buffering for header skipping
+	bufferSize := 2 * 1024 * 1024 // 2 MB buffer
+	if c.bufferSize > 0 {
+		bufferSize = c.bufferSize
+	}
+
+	counter := &CountReader{Reader: reader}
+	bufferedReader := bufio.NewReaderSize(counter, bufferSize)
+
+	switch {
+	case c.useColumnMapping:
+		if c.skip != 1 {
+			return Result{}, fmt.Errorf("column mapping requires skip to be exactly 1 (one header row)")
+		}
```

Comment:
As discussed during a pairing review session, If we want to keep letting the user skip **lines** at the beginning of the file, this breaks with such a feature. 
An alternative to this plus a test with a valid use case can be found in my draft pr: https://github.com/timescale/timescaledb-parallel-copy/pull/120
Also note there is a TODO we might want to discuss